ts/main.ts

```diff
@@ -46,3 +130,166 @@ function testKey(key: string): boolean {
 function testKeyStrict(key: string): boolean {
   return /^[A-Za-z0-9 ]$/.test(key);
 }
+
+function getRequestUrl(
+  option: string,
+  keyword: string,
+  url: string = 'https://api.datamuse.com/words?',
+): string {
+  //  Autocomplete suggestions only
+  if (option.includes('autocomplete')) {
+    url = `https://api.datamuse.com/sug?s=${keyword}`;
+  } else {
+    if (url !== 'https://api.datamuse.com/words?') {
+      if (url.includes('?')) {
+        if (!url.endsWith('&')) {
+          url += '&';
+        }
+      }
+    }
+    //  Root options
+    if (option.includes('meansLike')) {
+      url += `ml=${keyword}&`;
+    }
+    if (option.includes('soundsLike')) {
+      url += `sl=${keyword}&`;
+    }
+    if (option.includes('spelledLike')) {
+      url += `sp=${keyword}&`;
+    }
+    if (option.includes('followsWords')) {
+      url += `lc=${keyword}&`;
+    }
+    if (option.includes('followedWith')) {
+      url += `rc=${keyword}&`;
+    }
+    if (option.includes('startsWith')) {
+      url += `sp=${keyword}*&`;
+    }
+    if (option.includes('endsWith')) {
+      url += `sp=*${keyword}&`;
+    }
+    //  Related word options
+    if (option.includes('describesWords')) {
+      url += `rel_jja=${keyword}&`;
+    }
+    if (option.includes('describedWith')) {
+      url += `rel_jjb=${keyword}&`;
+    }
+    if (option.includes('associatedWith')) {
+      url += `rel_trg=${keyword}&`;
+    }
+  }
+  //  Trim last &amp;
+  if (url.endsWith('&')) {
+    if (url.includes('max')) {
+      url = url.substring(0, url.length - 1);
+    } else {
+      url += 'max=10';
+    }
+  }
+  return url;
+}
+
+async function makeRequest(url: string): Promise<ResponseWord[]> {
+  let responseArr = [] as ResponseWord[];
+  try {
+    const response = await fetch(url);
+    try {
+      if (!response) throw new Error();
+      try {
+        if (response.ok !== true) throw new Error();
+        responseArr = await response.json();
+      } catch (e) {
+        try {
+          console.log(
+            `Endpoint exists but cannot return page contents, error code (page status): ${
+              response.status
+            }. Full page response is ${response.json()}.`,
+            e,
+          );
+        } catch (e) {
+          console.log(
+            `Endpoint exists but cannot return page, nor can log see response.`,
+          );
+        }
+      }
+    } catch (e) {
+      console.log('Fetch succeeded but got no response.', e);
+    }
+  } catch (e) {
+    console.log('Endpoint does not exist or fetch failed.', e);
+  }
+  return responseArr;
+}
```

Comment:
I would refactor this into a single `try/catch` block rather than nesting multiple. By throwing an error instead, you can already change what message gets thrown to the `catch` statement.

```typescript
async function makeRequest(url: string): Promise<ResponseWord[]> {
  let responseArr = [] as ResponseWord[];
  try {
    const response = await fetch(url);
    if (!response) {
      throw new Error('Fetch succeeded but got no response.');
    }
    if (response.ok !== true) {
      throw new Error(
        `Endpoint exists but cannot return page contents, error code (page status): ${
          response.status
        }. Full page response is ${response.json()}.`,
      );
    }
    responseArr = await response.json();
  } catch (e) {
    console.log(e.message || 'Endpoint does not exist or fetch failed.', e);
  }
  return responseArr;
}
```